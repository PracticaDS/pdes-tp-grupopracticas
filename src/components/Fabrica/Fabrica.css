.Fabrica {
  display: grid;
  background-color: white;
}