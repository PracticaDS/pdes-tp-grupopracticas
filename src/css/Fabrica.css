.grilla {
    display: grid;
    grid-template-columns: auto;
    background-color: white;
  }