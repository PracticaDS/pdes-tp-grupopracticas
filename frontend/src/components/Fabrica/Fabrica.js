import React, { Component } from 'react'
import './Fabrica.css'
import Board from '../Board/Board'
import ToolBox from '../ToolBox/ToolBox'
import MachineDescription from '../MachineDescription/MachineDescription'


export default class Fabrica extends Component {

  render() {
    return (
      <div className="Fabrica">
        <ToolBox /> 
        <Board />
        <MachineDescription />
      </div>
    )
  }

}