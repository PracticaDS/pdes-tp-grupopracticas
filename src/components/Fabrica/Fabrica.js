import React, { Component } from 'react';
import './Fabrica.css';
import { PropTypes } from 'prop-types';
import { Fila } from '../Fila/Fila'

export class Fabrica extends Component {
  
  constructor(props) {
    super(props);
    this.state = {
      grillas: []
    }
  }

  render() {
    return (
        <div className="grilla">
            {this.dibujarTablero()}
        </div>
    )
  }
  
  dibujarTablero(){
    let grilla = [];
    for (let i = 1; i <= this.props.filas; i++) {
      grilla.push(
        <Fila key={i} index={i} columnas={this.props.columnas} />
      );
    }
    this.grillas = grilla;
    return grilla;
  }

}

Fabrica.propTypes = {
  filas : PropTypes.number,
  columnas: PropTypes.number
}

Fabrica.defaultProps = { 
  filas : 10,
  columnas: 10
}