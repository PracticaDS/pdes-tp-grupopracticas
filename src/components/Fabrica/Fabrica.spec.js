import React from 'react';
import ReactDOM from 'react-dom';
import Fabrica from './Fabrica'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Fabrica />, div);
  ReactDOM.unmountComponentAtNode(div);
});