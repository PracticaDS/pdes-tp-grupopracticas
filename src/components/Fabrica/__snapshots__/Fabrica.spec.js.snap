// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fabrica should render correctly whithout props 1`] = `
<div
  className="Fabrica"
>
  <Connect(Board) />
</div>
`;
